### ISSUE TITLE
mlDeploy unable to handle misconfigured module database

### ISSUE THREAD

**Issue by kgmanuel** [marklogic/ml-gradle#660]
**Describe the bug**
If the app server does not use a modules database, an mlDeploy or mlRedeploy will not succeed

**To Reproduce**
1. create a new project using `gradle mlNewProject`. supply 8100 for rest port.
2. deploy: `gradle mlDeploy`
3. Using admin UI, change the app server's module database to filesystem
4. deploy: `gradle mlDeploy` (this will fail)
5. re-deploy: `gradle mlReDeploy` (this will also fail)

On both 4 & 5 the error would be XDMP-UPDATEFUNCTIONFROMQUERY, which is also misleading.

**Expected behavior**
Basically an "undo" of #3 action above.

**Comment by rjrudin** [comment]
I was able to reproduce this, and I believe this is expected behavior, assuming that you have files under src/main/ml-modules that are associated with a REST API server. By changing the modules database to the filesystem, I believe your REST API server is no longer valid in that it's not using a proper modules database. So when ml-gradle deploys the src/main/ml-modules/options/myApp-options.xml file that is generated by `mlNewProject`, MarkLogic throws an error because it is unable to write that options file. 

Note that ml-gradle is just the messenger here - it has nothing to do with the failure nor the error message. I agree that the error message is not helpful, but that would need to be resolved via a bug report against the REST API.
